Fix: iterating over None when running on test splits.

tjelinek opened the pull request:
For test split, frame_objects is None (no annotations). At least for sequences on BOP huggingface. Previously caused error due to iterating over None.

Commit:
fix bug in generating statistics for Box2dDataProvider

Summary: Fix bug in generating statistics for Box2dDataProvider. We were using sum instead of len.

Reviewed By: SeaOtocinclus

Differential Revision: D61223159

fbshipit-source-id: 9dc35e8c2b8cdecc397117f9c59b6a8cdcbecd07

thodan committed:
Adding link to HOT3D homepage to README.md

Summary: Adding link to HOT3D homepage to README.md.

Reviewed By: SeaOtocinclus

Differential Revision: D61326997

fbshipit-source-id: fc83bf8b8ae744641767bac82705c9838da07220

thodan committed:
Updating number of HOT3D clips

Summary: Updating the number of clips after QA

Reviewed By: SeaOtocinclus

Differential Revision: D61382460

fbshipit-source-id: c4368f57ef62a7316cbb36a372dd32af3707a03f

SeaOtocinclus committed:
{Documentation} Add a version file to document dataset update

Summary: Introduce a VERSIONS.md file to document the dataset update and how to identify the version of the HOT3D sequence a user has downloaded.

Reviewed By: thodan

Differential Revision: D61359569

fbshipit-source-id: ca8e02f201d61b8f1790e5e96ff70fd56a4e9c88

thodan committed:
TDG from HOT3D-Clips

Summary: Adding a script for generating Torch datasets from HOT3D-Clips (that are saved in tar archives) + necessary changes.

Reviewed By: nv-nguyen

Differential Revision: D60901446

fbshipit-source-id: bea0d09d18c08297f434045bc1428ab3e105a307

thodan committed:
Adding support for loading and visualizing masks

Summary: This diff brings support for saving, loading and visualizing amodal and modal masks from HOT3D-Clips.

Reviewed By: prithvirb

Differential Revision: D61715780

fbshipit-source-id: 28dc96237c1037c7c68fd656240544f8a43dcca8

Commit:
add the SAM2_COMPUTE_SCORE step to flow job

Summary: add the SAM2_COMPUTE_SCORE step to flow job

Reviewed By: SeaOtocinclus

Differential Revision: D61341812

fbshipit-source-id: b8ccfe38d5e9ff2bd5068843b18468dbf6422221

Commit:
removing base_module] fbcode/nimble/opensource/hand_tracking_toolkit/hand_tracking_toolkit/TARGETS

Differential Revision: D62122885

fbshipit-source-id: 261cc68c0a0e9e11c7472485b2b137218bee46d0

lg-zhang committed:
Back out "removing base_module] fbcode/nimble/opensource/hand_tracking_toolkit/hand_tracking_toolkit/TARGETS"

Summary:
Original commit changeset: 261cc68c0a0e

Original Phabricator Diff: D62122885

We have to do use base_module to make the library usable in both fbcode and OSS.

Differential Revision: D62254688

fbshipit-source-id: 39b5f67385f64508099de8f8473db2b0fbebfb02

annakukliansky committed:
refactor s*

Summary: The logic of parutil.get_file_path can be replaced with python implementation via importlib. Thus we wont need to support proprietary code: easier to open source, less prone to bugs, less dependencies in buck targets, etc .

Reviewed By: markisaa

Differential Revision: D62052032

fbshipit-source-id: acd6fa2d4b4e0a15b4e5a1635582cf3b335afe7d

Commit:
removing base_module] fbcode/nimble/opensource/hand_tracking_toolkit/hand_tracking_toolkit/TARGETS

Reviewed By: neccam-meta

Differential Revision: D62434873

fbshipit-source-id: bb8a7c66c521b206acb9eb1f61d624afe94a86f1

Commit:
Revert D62434873

Summary:
This diff reverts D62434873
(The context such as a Sandcastle job, Task, SEV, etc. was not provided.)

Reviewed By: lg-zhang

Differential Revision: D62483639

fbshipit-source-id: 078cefc2f834f971c8434f18d0af6faaca06128e

SeaOtocinclus committed:
{Documentation} Fix syntax

Summary: Fix markdown syntax and some typos

Reviewed By: thodan

Differential Revision:
D62253894

Privacy Context Container: L1226793

fbshipit-source-id: 6dab2e130893eb322870080e4924c6c06e92868a

AnasIbrahim committed:
Add conversion code to the BOP format (#12)

Summary:
This PR contains 3 scripts to convert the HOT3D-Clips dataset from its native format to the BOP format. The BOP format had to be extended because the original format only included RGB and depth modalities. This extension made it possible to include a varying number of modalities (RGB, gray1, gray2) with separate annotations for each modality. The dataset has already been converted using these scripts and uploaded to the [BOP website](https://bop.felk.cvut.cz/datasets/). These scripts went through several revisions, as the conversion process required multiple changes to adapt the dataset to the new standard format, which was initially not strictly defined and required frequent adjustments. The scripts in this PR are isolated and do not make any changes to the existing code in the repository.

The 3 Scripts are:
1. hot3d_models_to_bop.py : Script that converts the 3D object (full) models from the GLTF format to the PLY format
2. hot3d_models_eval_to_bop.py: Script to convert the eval object model
3. hot3d_clips_to_bop_scenewise.py: Script to convert the dataset itself

Pull Request resolved: https://github.com/facebookresearch/hot3d/pull/12

Reviewed By: prithvirb

Differential Revision: D62526222

Pulled By: SeaOtocinclus

fbshipit-source-id: 8ff1cb1f0368d91d7b1ad63300ecf7e9146cdb5d

Commit:
fbcode//surreal/hot3d/hot3d_oss/hot3d/data_loaders/tests [A]

Differential Revision: D62884251

fbshipit-source-id: 372b4bc1a094de3787c68974bada19efc596d0ad

Commit:
fbcode//surreal/hot3d/hot3d_oss/hot3d/data_loaders/tests [B]

Differential Revision: D62884366

fbshipit-source-id: 865621404efd7c839d41a6700e48ad0ce7f0c246

thomaspolasek committed:
Convert directory fbcode/surreal to use the Ruff Formatter

Summary:
Converts the directory specified to use the Ruff formatter in pyfmt

ruff_dog

If this diff causes merge conflicts when rebasing, please run
`hg status -n -0 --change . -I '**/*.{py,pyi}' | xargs -0 arc pyfmt`
on your diff, and amend any changes before rebasing onto latest.
That should help reduce or eliminate any merge conflicts.

allow-large-files

Reviewed By: zertosh

Differential Revision: D63858542

fbshipit-source-id: 302f3632afaf8b16bc434cd8d8ededc75aa70938

SeaOtocinclus committed:
{Notebook} - Add a section for hand vertices reprojection #14

Summary: Add a new section to show how to reprojection hand vertices (landmarks or mesh) to the native FishEye images

Reviewed By: prithvirb

Differential Revision: D63988469

fbshipit-source-id: 6ee354777ae423cc2fcd62c1b38de1ce363efa02

SeaOtocinclus committed:
Refactor vis_clip console progress update

Summary:
Refactor vis_clip to use tqdm to show progress report on:
- loading the glb models
- exporting clips
- exporting frames of a given clip

{BugFix} Enable to export CLIPS for umetrack hands even if MANO hand model are not loaded.

Reviewed By: prithvirb

Differential Revision: D64264666

fbshipit-source-id: 949221086a73ca2b8ee0733c662a7db575d68f46

annakukliansky committed:
refactor get_file_path in s-z/*

Summary: The logic of parutil.get_file_path can be replaced with python implementation via importlib. Thus we wont need to support proprietary code: easier to open source, less prone to bugs, less dependencies in buck targets, etc .

Reviewed By: athmasagar, markisaa

Differential Revision: D62773477

fbshipit-source-id: 73433ca921ee975dcd6ae36993acbf60512a75a2

thodan committed:
Scripts for HOT3D-Clips

Summary: Code related to HOT3D-Clips.

Reviewed By: SeaOtocinclus

Differential Revision: D57783589

fbshipit-source-id: 9d1c01ae9444a30115ae028297ea72971a2c9efe

SeaOtocinclus committed:
{BugFix} Return correct relative transform if model is rectilinear

Summary:
Fix the T_device_camera transform to be the correct one.
The process of creating the rectilinear/pinhole camera was overwritting it with a Identity transform, we are here saving the transform before and sending it back.

Reviewed By: prithvirb

Differential Revision:
D64411688

Privacy Context Container: L1226793

fbshipit-source-id: d63844c8ed0356888048aaf31ec8b287d1f76392

SeaOtocinclus committed:
{Code Sample} Offscreen rendering using pyrender

Summary:
Introduce a code sample to perform offscreen rendering using PyRender.
- Use PyRender Scene to setup:
  - 3D object meshes and poses
  - Rectilinear cameras corresponding to Aria Glasses camera poses
  - Hand mesh
- Perform offscreen rendering to retrieve COLOR/DEPTH/Segmentation masks
- Show how to map the rectilinear image back to the native FishEye camera model (do note that we are loosing some field of view)

Example of usage:
1. install deps
`pip install pyrender trimesh`
2. Run the code sample
`python render_3.py`

See the rendered images in the folder

Reviewed By: prithvirb

Differential Revision: D64411691

fbshipit-source-id: 5c5ab2e42ce6ab9c67c47c9cbe9b3449e8d1cbff

Commit:
Remove default typing argument in PACKAGE file] [batch:36/104] [shard:2/N]

Reviewed By: migeed-z

Differential Revision: D66310645

fbshipit-source-id: ad4cff5f371e41e2e9e941ac699df14120740852

Commit:
Revert D66310645

Summary:
This diff reverts D66310645
Due to C++ bindings, we get too many type errors that litter OSS. We will revert this diff and manually look into a better way to migrate it. We can also disable typing for that particular directory.

Reviewed By: migeed-z

Differential Revision: D66382862

fbshipit-source-id: bf3ee75a5dfce40652e4abb062df7c17529eb910

thodan committed:
Updating paper links

Summary: Changing links from HOT3D whitepaper to the new full paper submitted to CVPR and arXiv.

Reviewed By: SeaOtocinclus

Differential Revision: D66654050

fbshipit-source-id: 7d07177a85ef094a59e8a3ca3979ba8d611f8b04

SeaOtocinclus committed:
{Documentation} Details on HOT3D v3

Summary:
Update v3 in the dataset versions file
- Updated visibility scores in box2d_hands/objects.csv

Reviewed By: maieneuro

Differential Revision: D67524719

fbshipit-source-id: 5bb20b49011913f139982086daa4a682cd4825cc

YipuZhao committed:
{Feature} - AriaDataProvider - _timestamp_convert device <==> time code conversion

Summary: Enable device time to time code conversion

Reviewed By: SeaOtocinclus

Differential Revision: D68846880

fbshipit-source-id: 0b72ac8ec23f2c8dcbd22062739ffb117040705f

YipuZhao committed:
{BugFix} Rerun - Fix logging of eye gaze arrow

Summary: rerun [Arrows3D](https://ref.rerun.io/docs/python/0.21.0/common/archetypes/#rerun.archetypes.Arrows2D) takes `vectors` input arg which is agnostic from `origins`, hence the substraction.

Reviewed By: SeaOtocinclus

Differential Revision: D68743774

fbshipit-source-id: fded877d6a2d5cc2b45000f046d4f43f41a3b67a

tjelinek committed:
Fix: iterating over None when running on test splits.

For test split, frame_objects is None (no annotations). At least for sequences on BOP huggingface. Previously caused error due to iterating over None.

tjelinek committed:
Merge pull request #1 from tjelinek/fix_test_split_conversion

Fix: iterating over None when running on test splits.

AnasIbrahim commented:
Hi,
@tjelinek, Can you explain the issue a bit more please. I guess it is better to actually open a pull request on hot3d original repo. Just mention me there and I can help.
@nv-nguyen I think the link on the bop website (datasets page) get people to this fork. We should refer to the original [hot3d repo converter folder](https://github.com/facebookresearch/hot3d/tree/main/hot3d/clips/bop_format_converters). So please remove my name and the link to this fork from the bop website and just refer to the hot3d repo.

tjelinek commented:
@AnasIbrahim  Ok, first to the issue. I can then open a pull request; The problem is that when you want to convert test_aria/test_quest3d, this [line](https://github.com/AnasIbrahim/hot3d/blob/de7666d6659cd4c078f603ffccdf133df0039c40/hot3d/clips/bop_format_converters/hot3d_clips_to_bop_scenewise.py#L196) fails because this [function](https://github.com/AnasIbrahim/hot3d/blob/de7666d6659cd4c078f603ffccdf133df0039c40/hot3d/clips/clip_util.py#L96) returns None on test sequences (they do not contain some of the ground-truth data). I downloaded the data from bop [huggingface](https://huggingface.co/datasets/bop-benchmark/hot3d/tree/main).

Thank you for pointing out that it is already merged with the original hot3d repository; I did not expect it.

@AnasIbrahim  @nv-nguyen Since the scene-wise conversion script in the original hot3d repo does not work on the bop sequences from huggingface, but for a different reason than above, I would first like to ask you if the intended usage is on the data downloaded as described in hot3d README or from bop huggingface? Then it may be a good idea to update instructions at the bop huggingface, the website, and in the conversion readme files accordingly.

AnasIbrahim commented:
@tjelinek thanks for the explanation. I see the mistake here. The "public" test scenes do not contain the annotation. I assumed when I wrote the script that the BOP team do the conversion to the BOP format and publish the test dataset without the annotations.

@nv-nguyen  @thodan is there a reason that we keep the hot3d with its native folder? is this just for preference? or could we just also host the converted dataset to the BOP format on huggingface? if we are not going host the converted data then I can help @tjelinek to make this change to the original hot3d repo.

@tjelinek if we make the changes i would first redo the commits from the original hot3d repo just to have an easier merge. I also would explicitly specify a flag as input argument in argparse (for example --test-annotations-exist default to false).

YipuZhao committed:
{BugFix} Handle invalid image in aria DataProvider

Summary: Handle case where VRS does not have an image for a given vrs recording index

Reviewed By: SeaOtocinclus

Differential Revision: D69128667

fbshipit-source-id: 9cb68124b5ce90e1a3926cc24f0098bd3ec6aec1

thodan commented:
@AnasIbrahim, it would be great if you could resolve this issue (so the script would be usable not only internally). Thanks!

@nv-nguyen is about to upload the HOT3D dataset converted to the BOP format to Hugging Face (should be done in 1--2 weeks).

SeaOtocinclus committed:
{Documentation} Fix HuggingFace HOT3D-Clips link

Summary: Fix invalid link to the HuggingFace dataset

Reviewed By: sindish

Differential Revision: D72391043

fbshipit-source-id: 41018332dc01341fb41f427b9ba46de3ce91811c

thodan committed:
Updating broken Hugging Face links

Summary: HOT3D-Clips were moved to a separate HF repo. Updating links.

Reviewed By: sindish

Differential Revision: D73022928

fbshipit-source-id: 09221eb1fee17601513041f2fbe6418ebb4f927e

tjelinek committed:
allow rewrite file

tjelinek committed:
Merge remote-tracking branch 'origin/main'

# Conflicts:
#	hot3d/clips/bop_format_converters/README.md
#	hot3d/clips/bop_format_converters/hot3d_clips_to_bop_scenewise.py
#	hot3d/clips/bop_format_converters/hot3d_models_eval_to_bop.py
#	hot3d/clips/bop_format_converters/hot3d_models_to_bop.py
